Source/items.cpp

```diff
@@ -2323,20 +2323,20 @@ int RndUItem(int m)
 		}
 	}
 
-	return ril[GenerateRnd(ri)];
+	return static_cast<_item_indexes>(ril[GenerateRnd(ri)]);

```

Comment:
Make the array use the item ID type as well

Reply:
Not sure this would be a lot better, as we'll just move the `static_cast` to the indexer.

I think it would be better to come back to this once we change how `AllItems` array is accessed (not via an array, preferably).